### PR TITLE
Replaced n_fold with n_folds

### PR DESCRIPTION
**Pull request by Saanidhyavats**
This PR solves  #7882

### CODE REVIEW

File: chainer/datasets/sub_dataset.py [hunk 1]
```diff
@@ -222,15 +222,15 @@ def get_cross_validation_datasets(dataset, n_fold, order=None):
     return splits
 
 
-def get_cross_validation_datasets_random(dataset, n_fold, seed=None):
+def get_cross_validation_datasets_random(dataset, n_folds, seed=None):
```
[the remark this answers is not in the document]
Comment (himkt) [reply chainer/datasets/sub_dataset.py]: If `n_fold` is specified, please notify that it is deprecated.
(https://github.com/chainer/chainer/blob/master/chainer/links/model/vision/resnet.py#L558-L563)

File: chainer/datasets/sub_dataset.py [hunk 2]
```diff
@@ -207,8 +207,8 @@ def get_cross_validation_datasets(dataset, n_fold, order=None):
         order = numpy.array(order)  # copy
 
     whole_size = len(dataset)
-    borders = [whole_size * i // n_fold for i in six.moves.range(n_fold + 1)]
-    test_sizes = [borders[i + 1] - borders[i] for i in six.moves.range(n_fold)]
+    borders = [whole_size * i // n_folds for i in six.moves.range(n_folds + 1)]
+    test_sizes = [borders[i + 1] - borders[i] for i in six.moves.range(n_folds)]
```
[the remark this answers is not in the document]
Comment (Saanidhyavats) [reply chainer/datasets/sub_dataset.py]: I will make the suggested changes.

File: chainer/datasets/sub_dataset.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Saanidhyavats) [reply chainer/datasets/sub_dataset.py]: I will make the suggested changes.

[the remark this answers is not in the document]
Comment (Saanidhyavats) [reply chainer/datasets/sub_dataset.py]: Shall I add same changes to `get_cross_validation_datasets` function

[the remark this answers is not in the document]
Comment (himkt) [reply chainer/datasets/sub_dataset.py]: Thank you for noticing me. Yes, please.

[the remark this answers is not in the document]
Comment (Saanidhyavats) [reply chainer/datasets/sub_dataset.py]: I have made the suggested changes, hope this resolves the issue

File: chainer/datasets/sub_dataset.py [hunk 3]
```diff
@@ -182,33 +182,45 @@ def split_dataset_n_random(dataset, n, seed=None):
...
         order = numpy.arange(len(dataset))
     else:
         order = numpy.array(order)  # copy
 
     whole_size = len(dataset)
-    borders = [whole_size * i // n_fold for i in six.moves.range(n_fold + 1)]
-    test_sizes = [borders[i + 1] - borders[i] for i in six.moves.range(n_fold)]
+    borders = [whole_size * i // n_folds for i in six.moves.range(n_folds + 1)]
+    test_sizes = [borders[i + 1] - borders[i]
+                  for i in six.moves.range(n_fold)]
```
Comment (himkt) [inline chainer/datasets/sub_dataset.py]:
```suggestion
                  for i in six.moves.range(n_folds)]
```

File: chainer/datasets/sub_dataset.py [hunk 4]
```diff
@@ -182,33 +182,45 @@ def split_dataset_n_random(dataset, n, seed=None):
...
         order (sequence of ints): Order of indexes with which each split is
             determined. If it is ``None``, then no permutation is used.
 
     Returns:
         list of tuples: List of dataset splits.
 
     """
+    
+    if 'n_fold' in kwargs:
+            warnings.warn(
```
Comment (himkt) [inline chainer/datasets/sub_dataset.py]: You need to import the module `warnings`

Comment (Saanidhyavats) [reply chainer/datasets/sub_dataset.py]: I am unable to understand the reason for error, can you(@himkt ) please further guide me on this issue.

File: chainer/datasets/sub_dataset.py [hunk 5]
```diff
@@ -182,33 +182,45 @@ def split_dataset_n_random(dataset, n, seed=None):
             for i in six.moves.range(n)]
 
 
-def get_cross_validation_datasets(dataset, n_fold, order=None):
+def get_cross_validation_datasets(dataset, n_folds=None, order=None,**kwargs):
```
Comment (himkt) [inline chainer/datasets/sub_dataset.py]: `E231 missing whitespace after ','`
Please add a whitespace before `**kwargs`.

File: chainer/datasets/sub_dataset.py [hunk 6]
```diff
@@ -182,33 +182,45 @@ def split_dataset_n_random(dataset, n, seed=None):
...
+             Please use `n_folds` instead.
+        n_folds (int): Number of splits for cross validation.
         order (sequence of ints): Order of indexes with which each split is
             determined. If it is ``None``, then no permutation is used.
 
     Returns:
         list of tuples: List of dataset splits.
 
     """
+
```
Comment (himkt) [inline chainer/datasets/sub_dataset.py]: `chainer/datasets/sub_dataset.py:207:1: W293 blank line contains whitespace`
Please remove whitespace in blank line.

File: chainer/datasets/sub_dataset.py (hunk 4 above)
Comment (himkt) [inline chainer/datasets/sub_dataset.py]: `chainer/datasets/sub_dataset.py:209:13: E117 over-indented`
Please set an indent width to `4`.

File: chainer/datasets/sub_dataset.py [hunk 7]
```diff
@@ -182,33 +182,45 @@ def split_dataset_n_random(dataset, n, seed=None):
...
     else:
         order = numpy.array(order)  # copy
 
     whole_size = len(dataset)
-    borders = [whole_size * i // n_fold for i in six.moves.range(n_fold + 1)]
-    test_sizes = [borders[i + 1] - borders[i] for i in six.moves.range(n_fold)]
+    borders = [whole_size * i // n_folds for i in six.moves.range(n_folds + 1)]
+    test_sizes = [borders[i + 1] - borders[i]
+                  for i in six.moves.range(n_folds)]
+
```
Comment (himkt) [inline chainer/datasets/sub_dataset.py]: `chainer/datasets/sub_dataset.py:225:5: E303 too many blank lines (2)`
Please remove `a` blank line. (multiple blank lines are not allowed.)

File: chainer/datasets/sub_dataset.py [hunk 8]
```diff
@@ -222,15 +234,18 @@ def get_cross_validation_datasets(dataset, n_fold, order=None):
     return splits
 
 
-def get_cross_validation_datasets_random(dataset, n_fold, seed=None):
+def get_cross_validation_datasets_random(dataset, n_folds, seed=None):
```
Comment (himkt) [inline chainer/datasets/sub_dataset.py]:
```
chainer/datasets/sub_dataset.py:259:20: F821 undefined name 'kwargs'
chainer/datasets/sub_dataset.py:264:23: F821 undefined name 'kwargs'
```

Please add keyword argument `**kwargs` to `get_cross_validation_datasets_random`.

File: chainer/datasets/sub_dataset.py [hunk 9]
```diff
@@ -241,5 +256,11 @@ def get_cross_validation_datasets_random(dataset, n_fold, seed=None):
         list of tuples: List of dataset splits.
 
     """
+    if 'n_fold' in kwargs:
+            warnings.warn(
```
Comment (himkt) [inline chainer/datasets/sub_dataset.py]: `chainer/datasets/sub_dataset.py:260:13: E117 over-indented`
Please set an indent width to 4.

**Comment by Saanidhyavats** [comment]
@himkt thanks for guiding me on the issue and sorry for inconvenience. I have updated the code, hope this resolves the issue

File: chainer/datasets/sub_dataset.py [hunk 10]
```diff
@@ -182,33 +182,44 @@ def split_dataset_n_random(dataset, n, seed=None):
...
     part to the test dataset. No pairs of test datasets share any examples, and
     all test datasets together cover the whole base dataset. Each test dataset
     contains almost same number of examples (the numbers may differ up to 1).
 
     Args:
         dataset: Dataset to split.
-        n_fold (int): Number of splits for cross validation.
+        n_fold(int): *(deprecated)*
+            `n_fold` is now deprecated for consistency of naming choice.
+             Please use `n_folds` instead.
```
Comment (himkt) [inline chainer/datasets/sub_dataset.py]:
```suggestion
            Please use `n_folds` instead.
```

Comment (himkt) [reply chainer/datasets/sub_dataset.py]:
```
chainer/datasets/sub_dataset.py:docstring of chainer.datasets.get_cross_validation_datasets:12:Unexpected 
indentation.
```

https://travis-ci.org/chainer/chainer/jobs/626155112?utm_medium=notification&utm_source=github_status

**Comment by himkt** [comment]
Jenkins, test this please.

**Comment by himkt** [comment]
@Saanidhyavats This PR was merged, thank you for contribution!

**Comment by Saanidhyavats** [comment]
Thank you for accepting my pull request.

**Pull request merged**